src/compiler/binder.rs

```diff
@@ -227,34 +227,26 @@ impl BinderType {
 
     fn bind_children(&self, node: &Node) {
         match node {
-            Node::Statement(statement) => match statement {
-                Statement::ExpressionStatement(expression_statement) => {
-                    self.bind_expression_statement(expression_statement);
-                }
-                _ => {
-                    self.bind_each_child(node);
-                }
-            },
-            Node::Expression(expression) => match expression {
-                Expression::PrefixUnaryExpression(_) => {
-                    self.bind_prefix_unary_expression_flow(node);
-                }
-                Expression::BinaryExpression(_) => unimplemented!(),
-                _ => {
-                    self.bind_each_child(node);
-                }
-            },
+            Node::Statement(Statement::ExpressionStatement(expression_statement)) => {
```

Comment:
Per a comment you'd made, I think "flattening" these node sub-variant `match` cases is a good idea for keeping the structure more similar to the Typescript version

In this case, that enabled just having a single "default" `self.bind_each_child(node);` at the end instead of having to nest it as a default inside each sub-variant